src/openrct2/rct2/S6Importer.cpp

```diff
@@ -919,7 +919,7 @@ class S6Importer final : public IParkImporter
     void ImportResearchList()
     {
         bool invented = true;
-        for (size_t i = 0; i < sizeof(_s6.research_items); i++)
+        for (RCT12ResearchItem& ri : research_items)
```

Comment:
```suggestion
        for (const auto& ri : research_items)
```

Reply:
Nice use of auto. Not sure if `const` can be used here, since a pointer gets passed in the body. Can't try it right now though.